addons/survey/models/survey_survey.py

```diff
@@ -563,6 +563,29 @@ def _get_next_page_or_question(self, user_input, page_or_question_id, go_back=Fa
 
         return Question
 
+    def _is_last_page_or_question(self, user_input, page_or_question):
+        """ This method checks if the given question or page is the last one.
+        This includes conditional questions configuration. If the given question is normally not the last one but
+        every following questions are inactive due to conditional questions configurations (and user choices),
+        the given question will be the last one.
+        For section, we check in each following section if there is an active question.
+        If yes, the given page is not the last one.
+        """
+        pages_or_questions = self._get_pages_or_questions(user_input)
+        current_page_index = pages_or_questions.ids.index(page_or_question.id)
+        next_page_or_question_candidates = pages_or_questions[current_page_index + 1:]
+        is_last_question = True
+        if next_page_or_question_candidates:
+            inactive_questions = user_input._get_inactive_conditional_questions()
+            if self.questions_layout == 'page_per_question':
+                is_last_question = not any(next_question not in inactive_questions for next_question in next_page_or_question_candidates)
```

Comment:
[C7] Why keeping a is_last_question variable ? You could make direct returns like
```
if not next_page_or_question_candidates: return False
...
return not any(blah) 
...
```

Just sayin'